### PR TITLE
Update index.php

### PR DESCRIPTION
**Pull request by WinterSilence**
`__DIR__` The directory of the file. If used inside an include, the directory of the included file is returned. This is equivalent to dirname(`__FILE__`). This directory name does not have a trailing slash unless it is the root directory. (Added in PHP 5.3.0.)

if ( ! defined('KOHANA_START_TIME'))
if ( ! defined('KOHANA_START_MEMORY'))

Why this check? and why not the same for APPPATH, MODPATH, ..?

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -54,7 +54,7 @@
  */
 
 // Set the full path to the docroot
-define('DOCROOT', realpath(dirname(__FILE__)).DIRECTORY_SEPARATOR);
+define('DOCROOT', rtrim(__DIR__, DIRECTORY_SEPARATOR).DIRECTORY_SEPARATOR);
```
Comment (shadowhand) [inline index.php]: you still need `realpath`, instead of `rtrim`.

File: index.php [hunk 2]
```diff
@@ -79,7 +79,8 @@
 if (file_exists('install'.EXT))
 {
 	// Load the installation check
-	return include 'install'.EXT;
+	include 'install'.EXT;
```
Comment (shadowhand) [inline index.php]: why this change?

File: index.php (hunk 1 above)
Comment (WinterSilence) [inline index.php]: in what cases `__ DIR__` contains no absolute path?

Comment (shadowhand) [inline index.php]: realpath also resolves symlinks. without it, we have run into errors with some server configurations.

File: index.php (hunk 2 above)
Comment (WinterSilence) [inline index.php]: It is considered bad style, in addition to other places, we also use `die()`

Comment (zombor) [inline index.php]: We only use `die` in one other place, afaik. `return` is correct here, I think.

File: index.php (hunk 1 above)
Comment (WinterSilence) [inline index.php]: thanks, got it. fix.

File: index.php (hunk 2 above)
Comment (WinterSilence) [inline index.php]: not in one: https://github.com/WinterSilence/kohana/blob/3.3/master/application/classes/Controller/Welcome.php#L1 , for me it's not so principal, will correct now.

**Comment by WinterSilence** [comment]
Also:

`if ( ! defined('KOHANA_START_TIME'))`
`if ( ! defined('KOHANA_START_MEMORY'))`
Why this check? and why not the same for APPPATH, MODPATH, ..?

Why uses memory_get_usage() rather than memory_get_usage(TRUE)?

**Comment by kemo** [comment]
Because `memory_get_usage(TRUE)` will not return the amount of memory we used but the amount PHP reserved for the process instead. Maybe it wouldn't be bad to have a benchmark for that as well?

**Comment by WinterSilence** [comment]
Thanks, understood. What about constants?

**Comment by shadowhand** [comment]
> `if ( ! defined('KOHANA_START_TIME'))`
> `if ( ! defined('KOHANA_START_MEMORY'))`
> Why this check?

Because someone might want to initialize their app before including Kohana and be more accurate about when the app started.

File: index.php [hunk 3]
```diff
@@ -79,7 +79,8 @@
...
+	die;
```
Comment (shadowhand) [inline index.php]: This needs to be reverted.

**Comment by WinterSilence** [comment]
And why then the this checks is done for paths? They may also be declared earlier - or check for the existence of all constants or not checked at all.

**Comment by shadowhand** [comment]
Because `APPPATH` and similar can be controlled by variables in index.php. It wouldn't make sense to set those before booting Kohana, whereas there are real world reasons for defining start time/memory before booting Kohana.

**Comment by WinterSilence** [comment]
All wishes were considered and fixed, we can add.
